src/openloco/window.cpp

```diff
@@ -700,6 +700,16 @@ namespace openloco::ui
         call((int32_t)this->event_handlers->viewport_rotate, regs);
     }
 
+    void window::call_text_input(widget_index caller, char* buffer)
+    {
+        registers regs;
+        regs.dx = caller;
+        regs.esi = (int32_t)this;
+        regs.cl = 1;
+        regs.edi = (uintptr_t)buffer;
+        call((uintptr_t)this->event_handlers->text_input, regs);
```

Comment:
Proxy event?

Reply:
It's not used in our code yet, but I added an assert